### ISSUE TITLE
Remove python-oauth from the archive

### ISSUE THREAD

**Issue by ubuntu-server-builder** [canonical/cloud-init#2358]
This bug was originally filed in Launchpad as [LP: #1118815](https://bugs.launchpad.net/cloud-init/+bug/1118815)
<details>
<summary>Launchpad details</summary>
<pre>
affected_projects = ['charm-hacluster', 'gwibber', 'maas', 'pyjuju', 'u1db', 'identicurse (Ubuntu)', 'jsonbot (Ubuntu)', 'python-django-piston (Ubuntu)', 'python-oauth (Ubuntu)', 'turpial (Ubuntu)', 'tweepy (Ubuntu)']
assignee = smoser
assignee_name = Scott Moser
date_closed = 2019-02-25T19:20:38.737090+00:00
date_created = 2013-02-07T22:51:12.747400+00:00
date_fix_committed = 2019-02-25T19:20:38.737090+00:00
date_fix_released = 2019-02-25T19:20:38.737090+00:00
id = 1118815
importance = medium
is_complete = True
lp_url = https://bugs.launchpad.net/cloud-init/+bug/1118815
milestone = None
owner = barry
owner_name = Barry Warsaw
private = False
status = fix_released
submitter = barry
submitter_name = Barry Warsaw
tags = ['server-next', 'u1-notrack']
duplicates = []
</pre>
</details>

_Launchpad user **Barry Warsaw(barry)** wrote on 2013-02-07T22:51:12.747400+00:00_

This bug tracks the removal of python-oauth from the archive.  (See also https://blueprints.launchpad.net/ubuntu/+spec/foundations-r-python3-oauth for additional details).

There are several very good reasons to remove python-oauth and port all reverse depends to python-oauthlib.

 * upstream oauth has been abandoned since 2009
 * upstream oauth only supports OAuth 1 (and probably not even the RFC 5849 standard)
 * upstream oauth only supports Python 2
 * upstream oauthlib is actively maintained
 * upstream oauthlib supports Python 2 and Python 3
 * upstream oauthlib supports RFC 5849 and the OAuth2 spec draft

As of yet, we cannot remove python-oauth because of existing reverse dependencies.  I'll add each of those as bug tasks to this one for tracking purposes.  When the time comes, I'll subscribe ~ubuntu-archive to the bug to do the final deed.

It will need to be blacklisted from Debian sync too.

In the meantime, *please* don't write any new code using python-oauth!  Use python-oauthlib.

http://pypi.python.org/pypi/oauthlib

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Scott Moser(smoser)** wrote on 2013-03-27T13:03:59.489574+00:00_

Just quickly googling to see how easy this would be, the python-oauthlib documentation [1] says:
  "If you then you should take a look at requests-oauthlib [2] which has several examples of how to use OAuth1 with requests."

Is there any plan to get requests_oauthlib ?  We've just moved cloud-init to requests and this would seem like the path of least resistance for anyone else using requests.

--
[1] https://oauthlib.readthedocs.org/en/latest/client.html
[2] https://github.com/requests/requests-oauthlib

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Julian Edwards(julian-edwards)** wrote on 2013-05-28T03:06:44.313130+00:00_

MAAS is blocked on python-django-piston getting fixed first.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Barry Warsaw(barry)** wrote on 2013-05-28T13:32:55.506291+00:00_

@Julian: https://bitbucket.org/jespern/django-piston/issue/231/switch-to-oauthlib

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Julian Edwards(julian-edwards)** wrote on 2013-05-28T23:40:08.245000+00:00_

Cheers Bazza.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Dimitri John Ledkov(xnox)** wrote on 2014-02-01T22:05:59.760605+00:00_

there is requests_oauthlib implementation both in python2 and python3.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Colin Watson(cjwatson)** wrote on 2018-04-27T13:46:06.396100+00:00_

I'm removing the lazr.restfulclient task from this bug; this is tracked in bug 1672458 instead.  (I'd have marked that bug as a duplicate of this one, but as is often the way with bugs with many tasks, Launchpad is having trouble with the number of structural subscriptions and is timing out on me.  More focused bugs are easier to deal with anyway.)

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Andreas Hasenack(ahasenack)** wrote on 2019-02-28T12:39:45.235756+00:00_

Regarding python-django-piston, there is no change yet. Upstream issue unanswered (https://bitbucket.org/jespern/django-piston/issues/231/switch-to-oauthlib), and no new upstream release since a long time. rmadison shows ubuntu has had version 0.2.3 since precise, all the way up to disco.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Adam Collard(adam-collard)** wrote on 2019-09-19T14:59:34.038866+00:00_

This bug has not seen any activity in the last 6 months, so it is being automatically closed.

If you are still experiencing this issue, please feel free to re-open.

MAAS Team

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Trent Lloyd(lathiat)** wrote on 2020-03-11T02:57:40.568115+00:00_

python-oauth is significantly broken in Focal due to trying to import cgi.parse_qs which moved for Python3.8. This breaks the MAAS CLI.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Launchpad Janitor(janitor)** wrote on 2020-06-01T10:44:18.091094+00:00_

Status changed to 'Confirmed' because the bug affects multiple users.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Launchpad Janitor(janitor)** wrote on 2020-06-01T10:44:18.091094+00:00_

Status changed to 'Confirmed' because the bug affects multiple users.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Launchpad Janitor(janitor)** wrote on 2020-06-01T10:44:18.091094+00:00_

Status changed to 'Confirmed' because the bug affects multiple users.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Launchpad Janitor(janitor)** wrote on 2020-06-01T10:44:18.091094+00:00_

Status changed to 'Confirmed' because the bug affects multiple users.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Launchpad Janitor(janitor)** wrote on 2020-06-01T10:44:18.091094+00:00_

Status changed to 'Confirmed' because the bug affects multiple users.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Paride Legovini(paride)** wrote on 2020-06-30T13:28:18.413073+00:00_

I updated some tasks:

- identicurse: removed from Groovy => Fix Released
- jsonbot: removed from Bionic => Fix Released
- python-django-piston: removed from Eoan => Fix Released
- turpial: removed from Xenial => Fix Releaed
- tweepy: dependency dropped in Xenial => Fix Released
- u1db: pkg dropped in Focal => Fix Released
- gwibber: removed from Xenial => Fix Released
- pyjuju/juju: removed from Bionic => Fix Released

According to

https://bugs.launchpad.net/ubuntu/+source/python-oauth/+bug/1883875

MAAS is not a blocker for the removal of python-oauth from Groovy. I think it should be Fix Released too.

I am not familiar with the OpenStack hacluster charm.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Alex Kavanagh(ajkavanagh)** wrote on 2021-02-23T16:48:30.158632+00:00_

Not a charm bug now.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Athos Ribeiro(athos-ribeiro)** wrote on 2022-01-26T21:54:27.228295+00:00_

There was another comment a while ago on https://bugs.launchpad.net/ubuntu/+source/python-oauth/+bug/1883875 with concerns on MASS still needing this package. It seems this is no longer the case. If so, we could proceed with removal on this one.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Athos Ribeiro(athos-ribeiro)** wrote on 2022-02-16T17:02:58.701364+00:00_

I verified reverse dependencies in jammy by running:

reverse-depends src:python-oauth

and

reverse-depends src:python-oauth -a source

Neither returned dependencies.

**Comment by ubuntu-server-builder** [comment]
_Launchpad user **Steve Langasek(vorlon)** wrote on 2022-02-22T00:17:41.860670+00:00_

Removing packages:
	python-oauth 1.0.1-6ubuntu1 in jammy
		python-oauth 1.0.1-6ubuntu1 in jammy amd64
		python-oauth 1.0.1-6ubuntu1 in jammy arm64
		python-oauth 1.0.1-6ubuntu1 in jammy armhf
		python-oauth 1.0.1-6ubuntu1 in jammy i386
		python-oauth 1.0.1-6ubuntu1 in jammy ppc64el
		python-oauth 1.0.1-6ubuntu1 in jammy riscv64
		python-oauth 1.0.1-6ubuntu1 in jammy s390x
		python3-oauth 1.0.1-6ubuntu1 in jammy amd64
		python3-oauth 1.0.1-6ubuntu1 in jammy arm64
		python3-oauth 1.0.1-6ubuntu1 in jammy armhf
		python3-oauth 1.0.1-6ubuntu1 in jammy i386
		python3-oauth 1.0.1-6ubuntu1 in jammy ppc64el
		python3-oauth 1.0.1-6ubuntu1 in jammy riscv64
		python3-oauth 1.0.1-6ubuntu1 in jammy s390x
Comment: (From Debian) RoQA; obsolete, replaced by python-oauthlib; Debian bug #1006229
Remove [y|N]? y
